### PR TITLE
[TS] Fix Accordion, AccordionPanel, Carousel and CheckBox extended types definition

### PR DESCRIPTION
**Pull request by Isaius**
#### What does this PR do?
This PR adds and exports <component>ExtendedProps declarations for the Accordion, AccordionPanel, Carousel and CheckBox components, forming part of issue #4998.

#### Where should the reviewer start?
The changes are in the type definition files, and are simple and do not have any dependencies from external files. As they are individual, the review can start for any of them.

#### What testing has been done on this PR?
Since it's type definition, no tests were added. All the existing tests passed locally.

#### How should this be manually tested?
Extending any of the changed components have to contain the respective extended `JSX.IntrinsicElements` 

#### Any background context you want to provide?
This changes were discussed in the #4978 

#### What are the relevant issues?
Progress track: #4998 
Discussion: #4978 

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No. Just all the components tracked in #4998 that now are fixed when extending. 

#### Is this change backwards compatible or is it a breaking change?
backwards compatible.

### CODE REVIEW

**Review by abnersajr — APPROVED** [review]

File: src/js/components/Accordion/index.d.ts [hunk 1]
```diff
@@ -2,16 +2,17 @@ import * as React from 'react';
...
   animate?: AnimateType;
   onActive?: (activeIndexes: number[]) => void;
   multiple?: boolean;
   messages?: { tabContents?: string };
 }
 
-declare const Accordion: React.FC<BoxProps &
-  AccordionProps &
-  JSX.IntrinsicElements['div']>;
+export type AccordionExtendedProps = AccordionProps &
```
Comment (ShimiSun) [inline src/js/components/Accordion/index.d.ts:13]: I think we should export an interface rather than a type.

**Review by ShimiSun — CHANGES_REQUESTED** [review]

File: src/js/components/Accordion/index.d.ts (hunk 1 above)
Comment (Isaius) [reply src/js/components/Accordion/index.d.ts:13]: That's a problem we discussed in the closed issue #4978 . `extends` only allow extend one item, components that extends more than that need a special form.
I suggested use a intermediate variable, something like that:

```ts
interface a extends b { }
export interface c extends a { }
```

But we agree that isn't good, so @abnersajr came with this approach using `type` and the `&`.

Comment (abnersajr) [reply src/js/components/Accordion/index.d.ts:13]: There is a way to do it.
Here is an example.
https://github.com/grommet/grommet/pull/5040/files#diff-5f4f62bcd0e71612e1588e56d33792b4a74c8e011fb6b5fe9b06bb8436ddc307R23

**Review by abnersajr — CHANGES_REQUESTED** [review]
Change to export interfaces.

File: src/js/components/Accordion/index.d.ts (hunk 1 above)
Comment (Isaius) [reply src/js/components/Accordion/index.d.ts:13]: sure, i'll change that

File: src/js/components/Accordion/index.d.ts [hunk 2]
```diff
@@ -2,16 +2,18 @@ import * as React from 'react';
 import { AnimateType } from '../../utils';
 import { BoxProps } from '../Box';
 
-export interface AccordionProps {
+export interface AccordionProps extends BoxProps {
```
[the remark this answers is not in the document]
Comment (Isaius) [reply src/js/components/Accordion/index.d.ts]: Done. But the branch is outdated and IDK how to do with it.
I've tried to update and... things went wrong. So, I changed back to the previous commit.

**Review by ShimiSun — APPROVED** [review]

**Pull request merged**